test/helper.py

```diff
@@ -38,21 +42,30 @@ def check_evaluation(
 ):
     """
     Helper function to test Mathics expression against
-    its results
+    its results.
 
     Compares the expressions represented by ``str_expr`` and  ``str_expected`` by
     evaluating the first, and optionally, the second. If omitted, `str_expected`
     is assumed to be `"Null"`.
 
-    to_string_expr: If ``True`` (default value) the result of the evaluation is
-                    converted into a Python string. Otherwise, the expression is kept
-                    as an Expression object.
-                    If this argument is set to ``None``, the session is reset.
+    str_expr: The expression to be tested. If its value is ``None``, the session is
```

Comment:
@rocky, when you call a `None` test, this function calls `reset_session`, and this is why it works.

Reply:
This is convoluted.